### PR TITLE
Add ESLint and Fix errors

### PR DESCRIPTION
**Pull request by JasonHoffmann**
This adds a basic ESLint config and addresses the main issues. This follows basic ESLint and React rules, so really the only things missing were semi-colons and Type checking. Those have both been added in.

Will need to explore ESLint configs that conform to WP standards a bit better.

### CODE REVIEW

**Comment by jjeaton** [comment]
@JasonHoffmann merge conflicts here.

**Comment by JasonHoffmann** [comment]
Yea agreed, going to bump it down to a warning

File: .scrutinizer.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ build:
             - gulp lint
             - ./bin/install-wp-tests.sh wordpress_test root ''
         override:
+            - eslint-run "assets/src/**"
```
Comment (JasonHoffmann) [inline .scrutinizer.yml]: I was actually going to ask. Not sure how to make it fail on errors and not warnings. Do you know? Or is that automatic?

**Comment by jjeaton** [comment]
there may be a setting. it will fail the build if it returns a non-zero
exit code. so there may be a config setting (there is for phpcs) where it
says to still return a zero if there are only warnings. not a huge deal, we
can fix warnings too. The primary concern is that the build fails if eslint
fails.

**Comment by JasonHoffmann** [comment]
Looks like that's what ESLint does by default (https://github.com/eslint/eslint/issues/858). I'll add it to the gulp task instead of standalone. Will need to do some testing as new commits come in probably.

**Pull request merged**